### PR TITLE
Adding HeBits torrent provider

### PR DESCRIPTION
**Pull request by smoresmores**
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: medusa/providers/torrent/html/hebits.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+                return items
+
+            # Skip column headers
+            for row in torrent_rows:
+                try:
+                    heb_eng_title = row.find('div', class_='bTitle').find(href=re.compile("details\.php")).find('b').get_text(strip=True)
+                    index = heb_eng_title.index('/')
+                    title = heb_eng_title[index+1:]
+
+                    download_url = urljoin(self.url, row.find('div', class_='bTitle').find(href=re.compile("download\.php"))['href'])
```
Comment (medariox) [inline medusa/providers/torrent/html/hebits.py]: Please use urljoin() after the ```if not all([title, download_url]):``` check further below.

File: medusa/providers/torrent/html/hebits.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+                    torrent_size = row.find('div', class_='bSize').get_text(strip=True)
+                    size = convert_size(torrent_size[5:], sep='') or -1
+
+                    item = {
+                        'title': title,
+                        'link': download_url,
+                        'size': size,
+                        'seeders': seeders,
+                        'leechers': leechers,
+                        'pubdate': None,
```
Comment (medariox) [inline medusa/providers/torrent/html/hebits.py]: Provider has no publishing date?

File: medusa/providers/torrent/html/hebits.py [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+            'password': self.password,
+        }
+
+        response = self.session.post(self.urls['login'], data=login_params)
+
+        if not response or not response.text:
+            log.warning('Unable to connect to provider')
+            return False
+
+        if response.text == 'OK':
```
Comment (medariox) [inline medusa/providers/torrent/html/hebits.py]: Are you sure this is what you want? That expects the response.text to be exactly ```OK``` -  if you want to check if the response contains ```OK``` you would do it like this:
```
if 'OK' in response.text:
    return True
```
Same about the rest below.

Comment (smoresmores) [reply medusa/providers/torrent/html/hebits.py]: Yes, this is the exact response. Tested and it's working.

File: medusa/providers/torrent/html/hebits.py [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+            return True
+        elif response.text == 'Banned':
+            log.warning('User {0} is banned from HeBits', self.userename)
```
Comment (medariox) [inline medusa/providers/torrent/html/hebits.py]: Small typo ```self.userename```.

File: medusa/providers/torrent/html/hebits.py [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+        :param data: The raw response from a search
+        :param mode: The current mode used to search, e.g. RSS
+
+        :return: A list of items found
+        """
+        items = []
+
+        with BS4Parser(data, 'html5lib') as html:
+            torrent_table = html.find('div', class_='browse')
+            torrent_rows = torrent_table('div', class_=re.compile("^line")) if torrent_table else []
```
Comment (medariox) [inline medusa/providers/torrent/html/hebits.py]: Please use single quotes everywhere (same for below).

File: medusa/providers/torrent/html/hebits.py [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+
+            # Continue only if at least one release is found
+            if len(torrent_rows) < 2:
+                log.debug('Data returned from provider does not contain any torrents')
+                return items
+
+            for row in torrent_rows:
+                try:
+                    heb_eng_title = row.find('div', class_='bTitle').find(href=re.compile("details\.php")).find('b').get_text(strip=True)
+                    title = heb_eng_title.split('/')[1].strip()
```
Comment (medariox) [inline medusa/providers/torrent/html/hebits.py]: Do you really need to strip() again here (the line above has already strip=True).

Comment (smoresmores) [reply medusa/providers/torrent/html/hebits.py]: This one is necessary. The first isn't so I deleted the strip=True.

**Pull request merged**